### PR TITLE
Fixes #220: More fully specify Sec-CH-UA-Platform-Version

### LINKED ISSUE #220: On Windows, platform version string should be sufficient to differentiate between the various Win10+ releases
Windows historically updated the major or minor version number with each significant OS release. As a result, the "Windows NT <major>.<minor>" component of the UA string was sufficient for sites to know how to send appropriate installers targeting each OS release.

As of Windows 10, though, Windows has essentially frozen the major and minor build number. As a result, this logic is no longer able to work across different Win10 releases even as we approach the 6th anniversary of the first Win10 release.

For OS X, we're already exposing a similar level of detail in terms of version numbers across releases.

With the UA string, it would have been unsafe from a compat perspective to add a third version component since it would have broken many existing UA string parsers. With UA Client Hints, we have an opportunity to add this functionality back.

There's obviously a bit more entropy here, but this version component has only been changing about twice a year and with the Windows install base it's not particularly unique. Since this isn't a header included by default, we can also consider it as an input into any future "privacy budget" schemes.

So, in short: I think we should have browsers expose major.minor.build in `Sec-CH-UA-Platform-Version` on Windows clients. e.g. instead of "10.0" it would say something like "10.0.19042" (for "20H2").

This is possibly not really an issue to resolve here on the spec, but if we have general agreement we could at have at least one of the examples demonstrate this.

### PR DESCRIPTION
**Pull request by erik-anderson**
Document expected version formats by platform based on the currently shipping implementation in Chromium.

On Windows, we diverge from the current shipping implementation by defining it in terms of the `Windows.Foundation.UniversalApiContract` per the discussion in #220.

***
<a href="https://pr-preview.s3.amazonaws.com/erik-anderson/ua-client-hints/pull/245.html" title="Last updated on Jun 30, 2021, 8:01 PM UTC (8eb970f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/ua-client-hints/245/20da7bc...erik-anderson:8eb970f.html" title="Last updated on Jun 30, 2021, 8:01 PM UTC (8eb970f)">Diff</a>

### CODE REVIEW

**Comment by erik-anderson** [comment]
@miketaylr I haven't installed the Bikeshed tooling so there may be some formatting things that need to be cleaned up.

**Comment by miketaylr** [comment]
> @miketaylr I haven't installed the Bikeshed tooling so there may be some formatting things that need to be cleaned up.

👍  Formatting looks good, per https://pr-preview.s3.amazonaws.com/erik-anderson/ua-client-hints/pull/245.html

File: index.bs
Comment (miketaylr) [inline index.bs:622]: Since this is used in 3 places, it's probably useful to pull out a `"Parse the platform returned version string"` `<dfn>` here (modulo a better name), which can take a `platformReturnedVersionString` as an argument (presumably https://infra.spec.whatwg.org/#strictly-split (on 0x2E), then do some kind of `parseInt` operation (maybe?), grab the first 3 ints or zeros otherwise). 

If it returns a list of the 3 version components we care about (or 0s), you could then set `platformVersionComponentList` to the return value in the next step.

File: index.bs
Comment (miketaylr) [inline index.bs]: (Is there a missing word here? Query the `UIDevice` string returned by? Or maybe just an expression I'm not familiar with)

File: index.bs
Comment (miketaylr) [inline index.bs:633]: nit: missing `API` at end of sentence?

**Review by miketaylr — CHANGES_REQUESTED** [review]
Thanks for the PR! Looks good, just a few suggested tweaks.

@erik-anderson also, please add your name to https://wicg.github.io/ua-client-hints/#ack (if you'd like).

File: index.bs
Comment (erik-anderson) [reply index.bs]: Oops-- yep!

File: index.bs
Comment (erik-anderson) [reply index.bs]: I tried to review other Bikeshed docs and didn't see much about parsing things-- I think it's likely fine to just call out that it must be an unsigned integer.

File: index.bs
Comment (erik-anderson) [reply index.bs]: I made the pointer more clear to the API that Chromium is using.

File: index.bs
Comment (erik-anderson) [reply index.bs]: I made the pointer more clear to the API that Chromium is using and read more like the iOS description.

**Comment by miketaylr** [comment]
> I believe this aligns with what's currently in Chrome Canary. Previously the platform version number format was a little all over the place which https://chromium-review.googlesource.com/c/chromium/src/+/2891972 recently changed to always report three version components.

Oh hey, I somehow missed that landing 😅 .

> Would you prefer me to update the Windows description to have three parts (with the 1st component being the UniversalApiContract version and the 2nd and 3rd being 0)?

I think there's a pretty good argument for consistency if we go that route. WDYT?

**Review by miketaylr — APPROVED** [review]
LGTM! I'll defer to you if you think we should also require 3 components for Windows (and hold off merging until you let me know). If so, we should update the "one to three version parts" for "Other platforms" to be consistent.

**Comment by erik-anderson** [comment]
@miketaylr I think consistency for parsing across platforms is reasonable. We'll still ensure reasonable version comparison checks on Windows by sticking with only filling in the significant most version component. Please merge if the latest commit looks good.

**Review by miketaylr — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
